### ISSUE TITLE
Uncaught RangeError

### ISSUE THREAD

**Issue by MarketingPip** [TranscribeJs/transcribe.js#3]
Getting error on older version of Chrome / Windows 7.

```js
Uncaught RangeError: WebAssembly.Memory(): could not allocate memory
```

I would assume this has to do with Symbol.dispose (which might need a polyfill which is available)

**Comment by thurti** [comment]
I don't understand why this error should occur from Symbol.dispose. Symbol.dispose is not used in the code. Or am I missing something? Have you tried to add the polyfill?

Which Chrome Version are you using?
What's the filesize of the audio file you tried to process?

**Comment by MarketingPip** [comment]
@thurti - the error occurs on init. And if you look into Symbol.dispose it might be able to shed some light on this.

**Comment by thurti** [comment]
I've tested the [example](https://examples.transcribejs.dev/examples/index.html) on Win7, Chrome 109 (latest for Win7). Everthing works as expected without errors. Does the example run on your machine?

**Comment by thurti** [comment]
I only get this kind of error when I use a large model and/or large audio files. But that's expected since Chrome has a hard limit on memory. For newer versions this is about 2GB. This may be lower for older Chrome versions.

I will close the issue, but feel free to reopen if there are any new findings. Please make sure to share detailed information on how to reproduce the bug (like browser, OS, model file, audio file).

**Comment by thurti** [comment]
Sorry, I don't know how to solve your problem. It seems like you already know the reason, but don't want to tell me. I don't get it.

**Comment by MarketingPip** [comment]
@thurti - it appears you are not releasing the memory etc...

**Comment by thurti** [comment]
Ok, now I get it. 

Yes, that could be true. I will have a look.

**Comment by MarketingPip** [comment]
@thurti - my apologizes too! Didn't mean to be vague etc - come off as a asshole. Was not my intention at all! 🤗

**Comment by MarketingPip** [comment]
@thurti - I am stupid too. Just to clarify I am on a 4GB of ram system (I know). 

I am looking through Mozilla [here](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/Memory/Memory)

and I am curious what is the max memory this is allowed to grow to? (is it greater then my system?). 

And if so - is 4GB of memory **required** to run this in browser....?

(I am not a WASM expert either - so do bare in mind with me). 

Also leaving this reddit comment here as I feel it might help you bark up right alley (if this is due to memory not being properly allocated based etc) - [reddit comment](https://www.reddit.com/r/WebAssembly/comments/garm9c/comment/fp5mq01/)

**Comment by thurti** [comment]
>and I am curious what is the max memory this is allowed to grow to? (is it greater then my system?).

The wasm build is currently configured to allocate 2GB on init [here](https://github.com/TranscribeJs/transcribe.js/blob/ab4177545136a4d015e45e93c78dd5af70863b27/shout.wasm/src/CMakeLists.txt#L29) and is not allowed to grow dynamically. So 2GB will be the max used memory.

If you tell me your browser version I can check if this causes the problem. Maybe it would be better to set this value to 1GB on init and enable `ALLOW_MEMORY_GROWTH`.


Technically the max allowed to _allocate_ is 4GB (in the link you've provided under `Exceptions - RangeError`). **But** there is also a limit set by the browser. (see answers [here](https://stackoverflow.com/questions/40417774/memory-limits-in-webassembly). So I would keep the max of 2GB.

>(I am not a WASM expert either - so do bare in mind with me).

Me neither, still learning.


I'm not sure what you're trying to achieve. But if you want to run this on an old machine beware that this will be veeeery slow (like real time or slower). If you don't necessarily have to run this in a browser then there are better solutions out there.

**Comment by MarketingPip** [comment]
@thurti - Chrome 109. (I am using out of date browser, since I can not update :( . 4GB of memory (so that explains why I am eating this up). I am curious, maybe if this is set lower / memory is configured to choose memory dynamically etc. With this work?

```js
  let ram  = navigator?.deviceMemory 
       if(ram){
      const bytes = ram * 1000000000; 
      ram = `${bytes} bytes`
       } 
```

Maybe if in browser grab memory like etc? And figure out a good ratio / memory needed etc based on device specs?

**Comment by thurti** [comment]
@MarketingPip I've reduced the initial amount of memory that gets allocated. Does the [example](https://examples.transcribejs.dev/examples/index.html) now work on your machine?

**Comment by MarketingPip** [comment]
@thurti - great job. (but no cigar) it looks like you are making progress tho!

Getting some issues now due to :
```
Wasm SIMD unsupported
```

May I suggest this [library](https://github.com/GoogleChromeLabs/wasm-feature-detect)


Then something like if no detection?

```
Module.simd = false;
```

I am NOT saying this will work - but I am just trying to help you bark up the right tree lol

**Comment by MarketingPip** [comment]
@thurti - and there you go!

![image](https://github.com/user-attachments/assets/f529b369-3aa2-4c38-9d59-f489d993ed1b)

:D

**Comment by MarketingPip** [comment]
@thurti - playing around on iOS older device not working - give me that source code and I'll debug it tonight and see what else can be done.

**Comment by thurti** [comment]
Thanks for testing.

I don't want to disable SIMD in general because it makes the processing 2-3x faster. I think I will publish an additional "non SIMD" version of the wasm build and add some hints in the docs on how to use it.

Regarding other bugs, please open a new issues so I can keep track of them.

**Comment by MarketingPip** [comment]
@thurti - possibly make a flag that chooses build to use? Instead of separating or as said - just detect if supported or not with module above. And fallback based on that.

**Comment by MarketingPip** [comment]
@thurti - I think it should be out of box to fallback to a method that is supported. Why would I as a developer want to handle this?

**Comment by thurti** [comment]
> Why would I as a developer want to handle this?

Because I like to have control over what a library includes/requires into my project. The `FileTranscriber` constructor takes the wasm `createModule` function as a parameter. This way the developer can decide which wasm build to use. I think that's a good compromise between flexibility and control.

I think we have different opinions on what a library should do and what not. So let's just agree to disagree.

> Another idea here too - would be cool down times / possibly putting this into a worker. Then we can handle real time transcribing (somewhat / as fast as model can do it - via microphone input / word by word etc)

There is an experimental stream example with mic input.

**Comment by MarketingPip** [comment]
@thurti - that is more than fair. I do not mean to be trying to piss on your garden. Just saying my two cents - I would prefer as a developer / my usage of this library to auto just fallback to a working option so my app doesn't stop working etc / more doc to me to figure out why the library I imported is not working on XYZ's device. 

And dope! Still hoping to grab your new build used in that ^ so I can do some debugging on older iOS safari device.

**Comment by thurti** [comment]
No problem. I appreciate the input.

I've just released v2.0.2, that should fix the issue.
